### ISSUE TITLE
Impossible to link an intervention with an invoice API REST

### ISSUE THREAD

**Issue by antho2305** [Dolibarr/dolibarr#32344]
### Bug

Impossible to link an intervention with an invoice and vice versa using the api rest I get a 200 response using the swagger or using PHP there’s no error but no link is created between them

### Dolibarr Version

20.0.2

### Environment PHP

8.1

### Environment Database

MariaDB

### Steps to reproduce the behavior and expected behavior

Link an intervention with an invoice using “linkObjectsIds” using api rest

**Comment by JonBendtsen** [comment]
Is it possible to link them in the GUI?

(edit, seems like it is possible)

**Comment by antho2305** [comment]
Yes it is possible and as I understand it should be possible with the API Rest but unfortunately it doesn’t whereas I get a 200 response so it is accepted but doesn’t link

**Comment by JonBendtsen** [comment]
have you tried using the new objectlinks endpoint?

<img width="860" height="224" alt="Image" src="https://github.com/user-attachments/assets/e2146ab4-1dbd-4a63-a526-d2723535509d" />

**Comment by pixodeo** [comment]
On 21.0 the intervention API `POST /interventions` (`htdocs/fichinter/class/api_interventions.class.php:191`) iterates over `request_data` and assigns each field straight to `$this->fichinter`. So sending `"linkedObjectsIds": {...}` sets `$this->fichinter->linkedObjectsIds`, but `Fichinter::create()` only calls `add_object_linked()` when `$this->origin && $this->origin_id` are set (see `htdocs/fichinter/class/fichinter.class.php:366`). The linkedObjectsIds property is never written to `llx_element_element`, which is why the call returns 200 but no link appears.

Other Dolibarr APIs (members, member types, workstation, webhook) just `unset($object->linkedObjectsIds)` in their cleanup helpers and do not write them either. So the gap is general, not specific to interventions.

@JonBendtsen pointed at the newer `objectlinks` endpoint, which is the right place to express "link object A and object B" in a generic way. The workaround on 21.0 today is:

1. Create the intervention via `POST /interventions` with `origin: "facture"` and `origin_id: <invoice id>` in the body. `Fichinter::create()` honours those and writes the link at creation time.
2. Or, after creating the intervention, do a follow-up call on whichever `/<entity>/{id}/links` style endpoint your version exposes (depends on the branch).

A proper fix that makes `linkedObjectsIds` work on POST would be a non-trivial addition (multiple targets, asymmetric source/target semantics, permissions checks) that goes beyond the scope of an atomic bug fix. Happy to revisit it once the team picks a direction.
